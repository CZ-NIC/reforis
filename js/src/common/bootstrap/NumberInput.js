/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import PropTypes from "prop-types";
import Input from "./Input";

const NumberInput = ({ ...props }) => <Input type="number" {...props} />;

NumberInput.propTypes = {
    /** Field label. */
    label: PropTypes.string.isRequired,
    /** Error message. */
    error: PropTypes.string,
    /** Help text message. */
    helpText: PropTypes.string,
    /** Number value. */
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
};

export default NumberInput;
