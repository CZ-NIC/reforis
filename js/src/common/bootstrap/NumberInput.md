Bootstrap component of number input with label with predefined sizes and structure for using in foris forms.

All additional `props` are passed to the `<input type="number">` HTML component.

```js
import {useState} from 'react';
const [value, setValue] = useState(42);

<NumberInput
    value={value}
    label="Some number" 
    helpText="Read the small text!"
    min='33'
    max='54'
    onChange={target => setValue(target.value)}
/>
```
