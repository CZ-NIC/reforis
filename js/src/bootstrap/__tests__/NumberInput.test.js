/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */


import renderer from 'react-test-renderer';
import NumberInput from '../PasswordInput';
import React from 'react';

describe('<PasswordInput/>', () => {
    it('Render number input', () => {
        const tree = renderer
            .create(
                <NumberInput
                    label="Test label"
                    checked
                    helpText="Some help text"
                    value={1123}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});