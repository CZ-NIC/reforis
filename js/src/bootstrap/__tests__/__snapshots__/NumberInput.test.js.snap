// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput/> Render number input 1`] = `
<div
  className="form-group row"
>
  <label
    className="form-control-label col-sm-2"
    htmlFor="1"
  >
    Test label
  </label>
  <div
    className="col-sm-9"
  >
    <input
      checked={true}
      className="form-control "
      id="1"
      type="password"
      value={1123}
    />
    <div
      className="invalid-feedback"
    />
    <small
      className="form-text text-muted"
    >
      Some help text
    </small>
  </div>
</div>
`;
